### ISSUE TITLE
Release apps won't run when AndroidLinkMode=None

### ISSUE THREAD

**Issue by pjcollins** [xamarin/xamarin-android#5274]
### Steps to Reproduce

1.  Add `<AndroidLinkMode>None</AndroidLinkMode>` to one of our .NET 6 samples and try to run it:
    `dotnet build -t:Run HelloAndroid.csproj -p:Configuration=Release`

The .apk that is installed is at least missing assemblies, and can't be launched:

```
Exec
    Assembly = Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
    CommandLineArguments = "C:\Program Files (x86)\Android\android-sdk\platform-tools\adb"  shell am start -n "com.microsoft.net5.helloandroid/"
    Parameters
        Command = "C:\Program Files (x86)\Android\android-sdk\platform-tools\adb"  shell am start -n "com.microsoft.net5.helloandroid/"
    Exception occurred while executing 'start':
    java.lang.IllegalArgumentException: Bad component name: com.microsoft.net5.helloandroid/
    	at android.content.Intent.parseCommandArgs(Intent.java:7599)
    	at com.android.server.am.ActivityManagerShellCommand.makeIntent(ActivityManagerShellCommand.java:338)
    	at com.android.server.am.ActivityManagerShellCommand.runStartActivity(ActivityManagerShellCommand.java:435)
    	at com.android.server.am.ActivityManagerShellCommand.onCommand(ActivityManagerShellCommand.java:186)
    	at android.os.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:98)
    	at android.os.ShellCommand.exec(ShellCommand.java:44)
    	at com.android.server.am.ActivityManagerService.onShellCommand(ActivityManagerService.java:10522)
    	at android.os.Binder.shellCommand(Binder.java:929)
    	at android.os.Binder.onTransact(Binder.java:813)
    	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:5027)
    	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2883)
    	at android.os.Binder.execTransactInternal(Binder.java:1159)
    	at android.os.Binder.execTransact(Binder.java:1123)
    Errors
        C:\Users\Peter\android-toolchain\dotnet\packs\Microsoft.Android.Sdk.windows-x64\11.0.100-ci.net5-runtime-tests.216\tools\Xamarin.Android.Application.targets(27,5): error MSB3073: The command ""C:\Program Files (x86)\Android\android-sdk\platform-tools\adb"  shell am start -n "com.microsoft.net5.helloandroid/"" exited with code 255. [C:\Users\Peter\Desktop\HelloAndroid\HelloAndroid.csproj]
```

I haven't yet dug into this, I'm filing it for future investigation. I think we may have some build ordering issue to sort out. Here's a binlog: [nolink.zip](https://github.com/xamarin/xamarin-android/files/5502541/nolink.zip)

**Comment by jonathanpeppers** [comment]
I think I found this bug on another PR:

https://github.com/xamarin/xamarin-android/pull/5035/files#diff-ae990f1a2052dc51966ff8bd3f1de810710e0d29972845a834f4a6f6bb039c22R139

```diff
-    <ItemGroup Condition=" '$(AndroidLinkMode)' != 'None' And '$(AndroidIncludeDebugSymbols)' != 'true'  ">
+    <ItemGroup Condition=" '$(AndroidLinkMode)' != 'None' or '$(AndroidIncludeDebugSymbols)' != 'true'  ">
```

I will probably split this change out into its own PR, as the `MSBuildDeviceIntegration` PR is big and has a little ways to go before we can merge it.
